draft-ietf-quic-transport.md

```diff
@@ -1482,78 +1508,83 @@ A stream in the "open" state may be used by both peers to send frames of any
 type.  In this state, a sending peer must observe the flow-control limit
 advertised by its receiving peer ({{flow-control}}).
 
-From this state, either endpoint can send a frame with the FIN flag set, which
-causes the stream to transition into one of the "half-closed" states.  An
-endpoint sending an FIN flag causes the stream state to become "half-closed
-(local)".  An endpoint receiving a FIN flag causes the stream state to become
-"half-closed (remote)"; the receiving endpoint MUST NOT process the FIN flag
-until all preceding data on the stream has been received.
-
-Either endpoint can send a RST_STREAM frame from this state, causing it to
-transition immediately to "closed".
+From this state, either endpoint can send a STREAM frame with the FIN flag set
+or a RST_STREAM frame.  These indicate the clean or abrupt termination of data
+flow on the stream, respectively.  In either case, this causes the stream to
+transition into one of the "half-closed" states.  An endpoint sending an FIN
+flag or a RST_STREAM frame causes the stream state to become "half-closed
+(local)".  An endpoint receiving a FIN flag or a RST_STREAM frame causes the
+stream state to become "half-closed (remote)".  The receiving endpoint MUST NOT
+process the FIN flag until all preceding data on the stream has been received,
+but MAY process a RST_STREAM frame immediately and discard any preceding data.
```

Comment:
I tried to make this change, but couldn't find a place that made sense.  This is talking about the logic of stream state transitions, not about the parsing of the frame itself.  Technically, the "MUST NOT process" really means "don't complete this state transition with outstanding data."  I reworded this section to talk about the transitions and not the processing of the frame, which I think works better.